tools/generate-sha256tree-tests.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+from hashlib import sha256
+from random import randbytes, choice, randint, seed
```

Comment:
It's tempting to use these nice functions from `random`. Be cautioned that they hide global state (which you correctly set with `seed` to make these tests deterministic). That's fine in this case because things are so constrained but I want to be sure you're aware that `random` also includes stateful `Random` objects that you could explicitly pass through the call stack that would make these functions more "pure" and not dependent on local state.

So yeah, this is fine, but I just wanted to pass on an annoyance I have with random number generator libraries: they often hide state in secret globals.